src/java.base/share/classes/java/util/zip/GZIPInputStream.java

```diff
@@ -75,9 +76,29 @@ private void ensureOpen() throws IOException {
      * @throws    IllegalArgumentException if {@code size <= 0}
      */
     public GZIPInputStream(InputStream in, int size) throws IOException {
-        super(in, in != null ? new Inflater(true) : null, size);
+        super(in, createInflater(in, size), size);
         usesDefaultInflater = true;
-        readHeader(in);
+        try {
+            readHeader(in);
+        } catch (IOException ioe) {
+            this.inf.end();
+            throw ioe;
+        }
+    }
+
+    /*
+     * Creates and returns an Inflater only if the input stream is not null and the
+     * buffer size is > 0.
+     * If the input stream is null, then this method throws a
+     * NullPointerException. If the size is <= 0, then this method throws
+     * an IllegalArgumentException
+     */
+    private static Inflater createInflater(InputStream in, int size) {
+        Objects.requireNonNull(in);
```

Comment:
Hello Lance, both the constructors of `GZIPInputStream` already state that they throw a NullPointerException when the input stream is null:

```
     * @throws    NullPointerException if {@code in} is null
```
It was already being thrown from within the super constructor. To prevent creation of a `Inflater` when `in` is null, I had to add this check here.